File: daa/structures/test/com/tw/vrc/lists/QueueTest.java
package com.tw.vrc.lists;

import org.junit.Test;

import static junit.framework.Assert.assertEquals;

public class QueueTest {
    @Test
    public void shouldWork() {
       Queue<String> q = new Queue<String>();
       q.enqueue("A");
       q.enqueue("B");
       q.enqueue("C");
       assertEquals("A",q.dequeue());
       assertEquals("B",q.dequeue());
       assertEquals("C",q.dequeue());
    }
}

File: algos/graphs/src/com/vijayrc/algos/queue/QueueTest.scala
package com.vijayrc.algos.queue

import org.scalatest.FunSuite

class QueueTest extends FunSuite{

  test("should work for a link based queue"){
    val queue:Queue = new QueueLinks
    for(i <- 1 to 5)queue.enqueue(i)
    for(i <- 1 to 2)queue.dequeue()
    queue.show()
  }

}
